src/app/features/tech-record/create/components/hydrate-new-vehicle-record/hydrate-new-vehicle-record.component.ts

```diff
@@ -39,13 +42,53 @@ export class HydrateNewVehicleRecordComponent {
     this.summary?.checkForms();
 
     if (!this.isInvalid) {
-      this.store.dispatch(createVehicleRecord());
-      this.actions$.pipe(ofType(createVehicleRecordSuccess), take(1)).subscribe(() => this.navigateBack());
+      this.technicalRecordService.editableVehicleTechRecord$
+        .pipe(
+          withLatestFrom(this.technicalRecordService.batchVehicles$),
+          take(1),
+          map(([record, batch]) => {
+            const vehiclesToCreate = record ? [record] : [];
+            batch?.forEach(v => {
+              const newVehicle = {
+                ...record!,
+                vin: v.vin,
+                primaryVrm: v.trailerId ? v.trailerId : null,
+                trailerId: v.trailerId ? v.trailerId : null
+              } as VehicleTechRecordModel;
+
+              vehiclesToCreate.push(newVehicle);
+            });
+            return vehiclesToCreate;
```

Comment:
A bit more simplified:
```suggestion
            const vehiclesToCreate = record ? [record] : [];
            return vehiclesToCreate.concat(
              batch.map(v => ({
                ...record!,
                vin: v.vin,
                primaryVrm: v.trailerId ? v.trailerId : null,
                trailerId: v.trailerId ? v.trailerId : null
              } as VehicleTechRecordModel))
            );
```